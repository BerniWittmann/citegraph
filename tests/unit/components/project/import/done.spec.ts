import { shallowMount, RouterLinkStub } from '@vue/test-utils'
import { i18n } from '../../../setupPlugins'

import ProjectImportDoneComponent from '@/components/project/import/Done.vue'

describe('components/project/import/Done.vue', () => {
  function getWrapper () {
    return shallowMount(ProjectImportDoneComponent, {
      i18n,
      stubs: {
        RouterLink: RouterLinkStub
      }
    })
  }

  it('renders', () => {
    const wrapper = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders a button to add more files', () => {
    const wrapper = getWrapper()
    const btn = wrapper.findAll('v-btn-stub').at(0)
    expect(btn.exists()).toBeTruthy()
    expect(btn.text()).toEqual('project.import.done.add_more')
    expect(wrapper.emitted('first-step')).toBeFalsy()

    btn.vm.$emit('click')

    expect(wrapper.emitted('first-step')).toBeTruthy()
  })

  it('renders a button link to the explore page', () => {
    const wrapper = getWrapper()
    const btn = wrapper.findAll('v-btn-stub').at(1)
    expect(btn.exists()).toBeTruthy()
    expect(btn.text()).toEqual('project.import.done.explore')
  })
})
