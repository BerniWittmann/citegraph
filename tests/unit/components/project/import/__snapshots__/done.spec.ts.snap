// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/project/import/Done.vue renders 1`] = `
<div class="project-import-done">
  <h1>project.import.done.title</h1>
  <div class="my-10 d-flex justify-space-around">
    <v-btn-stub color="warning" tag="button" activeclass="" large="true" rounded="true" type="button">
      project.import.done.add_more
    </v-btn-stub>
    <v-btn-stub color="primary" tag="button" activeclass="" large="true" rounded="true" type="button">project.import.done.explore</v-btn-stub>
  </div>
</div>
`;
